ngi_pipeline/engines/qc_ngi/workflows.py

```diff
@@ -126,23 +153,38 @@ def workflow_fastq_screen(input_files, output_dir, config):
     num_threads = config.get("qc", {}).get("fastq_screen", {}).get("threads") or 1
     subsample_reads = config.get("qc", {}).get("fastq_screen", {}).get("subsample_reads")
 
+    # Determine which files need processing
+    fastq_screen_output_file_tmpls = ("{}_screen.png", "{}_screen.txt")
+    fastq_to_analyze = set()
+    for elt in input_files:
+        # This may be a read pair
+        if type(elt) is list:
+            # fastq_screen uses the name of the first read of the pair for output files
+            elt = tuple(elt)
```

Comment:
I'm making this an elt because lists aren't hashable and so I could add them to the `set()`. Maybe I should have commented that.